doodler/components.py

```diff
@@ -299,34 +302,43 @@ def __init__(self, **params):
 
     @classmethod
     def from_folder(cls, imgs_folder, **params):
-        """Return a list of JPG and JPEG images in a folder (not recursively).
+        """Return a list of JPG, JPEG, TIF, or TIFF images in a folder (not recursively).
         """
-        jpegs = [
+        imfiles = [
             p
             for p in pathlib.Path(imgs_folder).iterdir()
-            if p.is_file() and p.suffix in ('.jpg', '.jpeg')
+            if p.is_file() and p.suffix.lower() in ('.jpg', '.jpeg', '.tif', '.tiff')
         ]
-        jpegs = sorted(jpegs)
+        imfiles = sorted(imfiles)
         input_image = cls(**params)
-        input_image.param.location.objects = jpegs
-        input_image.location = jpegs[0]
+        input_image.param.location.objects = imfiles
+        input_image.location = imfiles[0]
         return input_image
 
     @staticmethod
     def read_from_fs(path) -> np.ndarray:
-        """Read a JPEG as an Numpy array.
+        """Read tif or jpeg as an nd np array.
         """
-        img = PIL.Image.open(path)
-        nb_bands = len(img.getbands())
-        if nb_bands not in (3, 4):
-            raise ValueError(
-                f'The input image "{path}" has {nb_bands} channel(s), '
-                'only JPEG with 3 or 4 channels are supported.'
-            )
-        # Some JPEG files follow the CMYK model instead of RGB
-        if nb_bands == 4:
-            img = img.convert('RGB')
-        arr = np.array(img)
+        fn, ext = os.path.splitext(path)
+        print(ext)
+        if (ext.lower() == '.jpg') or (ext.lower() == '.jpeg'):
+            img = Image.open(path)
+            nbands = len(img.getbands())
+            # previously, code assumed 4 bands indicated CMYK, but band 4 can also be alpha, so:
```

Comment:
I remember checking how many bands could have a JPEG image and read that 4 bands meant CMYK (here: https://gdal.org/drivers/raster/jpeg.html#creation-options) but that source may be incomplete. I wasn't aware of the `mode` attribute which is indeed a better way anyway! You can remove the comment in the code, it's clear enough.